File: solutions/1342/main.cpp
#include <gtest/gtest.h> 

using namespace std; 

class Solution {
public:
    int numberOfSteps(int num) {
        int result = 0; 
        while (num > 0) {
            if (num & 1) num -= 1; 
            else num >>= 1; 
            result++;
        }
        return result; 
    }
};

TEST(Test1342, SimpleTest) {
    const int num = 14; 
    Solution s; 
    EXPECT_EQ(s.numberOfSteps(num), 6); 
}

int main() {
    testing::InitGoogleTest(); 
    return RUN_ALL_TESTS(); 
}

File: solutions/1342/main.rs
struct Solution {}

impl Solution {
    pub fn number_of_steps(num: i32) -> i32 {
        let mut result: i32 = 0; 
        let mut num: i32 = num; 
        while num > 0 {
            if num & 1 == 1 {
                num -= 1; 
            }
            else {
                num >>= 1; 
            }
            result += 1; 
        }
        return result; 
    }
}

fn main() {
    assert_eq!(Solution::number_of_steps(14), 6); 
}

